src/s1reader/s1_orbit.py

```diff
@@ -337,7 +399,267 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
+
+    # Parse the XML files
+    tree1 = ElementTree.parse(file1)
+    tree2 = ElementTree.parse(file2)
+
+    root1 = tree1.getroot()
+    root2 = tree2.getroot()
+
+    # Extract the Validity_Start and Validity_Stop timestamps from both files
+    start_time1 = get_dt(root1, "Validity_Start")
+    stop_time1 = get_dt(root1, "Validity_Stop")
+    start_time2 = get_dt(root2, "Validity_Start")
+    stop_time2 = get_dt(root2, "Validity_Stop")
+
+    # Determine the new Validity_Start and Validity_Stop values
+    new_start_dt = min(start_time1, start_time2)
+    new_stop_dt = max(stop_time1, stop_time2)
+
+    # Update the Validity_Start and Validity_Stop timestamps in the first XML
+    root1.find(".//Validity_Start").text = "UTC=" + new_start_dt.strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+    root1.find(".//Validity_Stop").text = "UTC=" + new_stop_dt.strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+
+    # Combine the <OSV> elements
+    list_of_osvs1 = root1.find(".//List_of_OSVs")
+    list_of_osvs2 = root2.find(".//List_of_OSVs")
+
+    merge_osv_list(list_of_osvs1, list_of_osvs2)
+
+    outfile = _generate_filename(file1, new_start_dt, new_stop_dt)
+    tree1.write(outfile, encoding="UTF-8", xml_declaration=True)
+    return outfile
+
+
+def merge_osv_list(list_of_osvs1, list_of_osvs2):
+    # Extract the UTC from the OSV of the first XML
+    osv1_utc_list = [datetime.datetime.fromisoformat(osv1.find('UTC').text.replace('UTC=',''))
+                     for osv1 in list_of_osvs1]
+
+    min_utc_osv1 = min(osv1_utc_list)
+    max_utc_osv1 = max(osv1_utc_list)
+
+    for osv in list_of_osvs2.findall("OSV"):
+        utc_osv2 = datetime.datetime.fromisoformat(osv.find('UTC').text.replace('UTC=',''))
+
+        if min_utc_osv1 < utc_osv2 < max_utc_osv1:
+            continue
+        list_of_osvs1.append(osv)
+
+    # sort the OSVs in the conatenated OSV
+    list_of_osvs1 = _sort_list_of_osv(list_of_osvs1)
+
+    # Adjust the count attribute in <List_of_OSVs>
+    new_count = len(list_of_osvs1.findall("OSV"))
+    list_of_osvs1.set("count", str(new_count))
+
+    return list_of_osvs1
+
+
+def _generate_filename(file_base: str, new_start: datetime.datetime, new_stop: datetime.datetime) -> str:
+    """Generate a new filename based on the two provided filenames.
+
+    Parameters
+    ----------
+    file_base : str
+        The name of one of the concatenated files
+    new_start : datetime
+        The new first datetime of the updated orbital elements
+    new_stop : datetime
+        The new final datetime of the updated orbital elements
+
+    Returns
+    -------
+    str
+        Generated filename.
+    """
+    product_name = Path(file_base).name
+    # >>> 'S1A_OPER_AUX_PREORB_OPOD_20200325T131800_V20200325T121452_20200325T184952'.index('V')
+    # 41
+    fmt = "%Y%m%dT%H%M%S"
+    new_start_stop_str = new_start.strftime(fmt) + "_" + new_stop.strftime(fmt)
+    new_product_name = product_name[:42] + new_start_stop_str
```

Comment:
The hard-coded number `42` seems dangerous to me. A way to make it a bit more robust would be to split the `product_name` using underscores `_`.

Reply:
Agreed. I've changed the logic here to find a index for `_V`, and make use of that to slice the string `product_name`